waddrmgr/db.go

```diff
@@ -606,14 +645,63 @@ func serializeChainedAddress(branch, index uint32) []byte {
 	return rawData
 }
 
-// deserializeImportedAddress deserializes the raw data from the passed address
-// row as an imported address.
-func deserializeImportedAddress(addressID []byte, row *dbAddressRow) (*dbImportedAddressRow, error) {
+// deserializeImportedP2PKHAddress deserializes the raw data from the passed address
+// row as an imported P2PKH address.
+func deserializeImportedP2PKHAddress(addressID []byte, row *dbAddressRow) (*dbImportedP2PKHAddressRow, error) {
+	// The serialized imported address raw data format is:
+	//   <encpkhashlen><encpkhash>
+	//
+	// 4 bytes encrypted pkhash len + encrypted pkhash +
+
+	// Given the above, the length of the entry must be at a minimum
+	// the constant value sizes.
+	if len(row.rawData) < 4 {
+		str := fmt.Sprintf("malformed serialized imported address for "+
+			"key %s", addressID)
+		return nil, managerError(ErrDatabase, str, nil)
+	}
+
+	retRow := dbImportedP2PKHAddressRow{
+		dbAddressRow: *row,
+	}
+
+	var offset uint32
+	pkHashLen := binary.LittleEndian.Uint32(row.rawData[offset : offset+4])
+	offset += 4
+	retRow.encryptedPKHash = make([]byte, pkHashLen)
+	copy(retRow.encryptedPKHash, row.rawData[offset:offset+pkHashLen])
+	offset += pkHashLen
+
+	return &retRow, nil
+}
+
+// serializeImportedP2PKHAddress returns the serialization of the raw data field for
+// an imported P2PKH address.
+func serializeImportedP2PKHAddress(encryptedPKHash []byte) []byte {
```

Comment:
Because this function only deals with P2PKH/Hash160-sized data, we can avoid the need to encode the length.  Instead, I would use `ripemd160.Size` throughout the serialization and deserialization code to refer the size of the hash.